Closed feedback session: warning message not sufficiently prominent #5271

YongJieYongJie opened the pull request:
Fixes #5271

YongJieYongJie commented:
Rewords message and adds bold style for emphasis.
Adds modal warning prompt to confirm user reads the message.

![image](https://cloud.githubusercontent.com/assets/14101781/15139107/aab4e2c6-16c6-11e6-9c0f-8264b2ff8665.png)

YongJieYongJie commented:
// currently working on passing the tests

damithc commented:
The modal window title can be more specific.

wkurniawan07 commented:
Some tests are failing, can you get them to pass on your local machine?

wkurniawan07 commented on src/main/webapp/WEB-INF/tags/shared/warningModal.tag:
```diff
@@ -0,0 +1,18 @@
+<%@ tag description="Modal prompt with bootstrap warning color" %>
+<%@ taglib tagdir="/WEB-INF/tags/shared/feedbackSubmissionEdit" prefix="feedbackSubmissionEdit" %>
```
I don't think this tag import is needed.

YongJieYongJie commented:
@wkurniawan07 Ready for review!

Changes:
- Improved the modal title
- Added test method to wait for and dismiss the warning modal
  - Updated tests to use the method where necessary
- Ran godmode

![image](https://cloud.githubusercontent.com/assets/14101781/15209405/640c0716-1863-11e6-9589-86c241c9c9fe.png)

wkurniawan07 commented:
There's a merge conflict here, I apologize beforehand if it's caused by me meddling with the js file when adding ESLint rules... :p

YongJieYongJie commented:
@wkurniawan07 Merge conflicts resolved!

wkurniawan07 commented:
Looks good!

damithc commented on src/main/java/teammates/common/util/Const.java:
```diff
@@ -1010,7 +1010,7 @@ public static final String getjQueryUiFilePath(boolean isDevServer) {
         public static final String FEEDBACK_RESPONSE_RECIPIENT_ALREADY_EXISTS = "Error trying to update recipient for response, as another response with the same recipient already exists.";
         
-        public static final String FEEDBACK_SUBMISSIONS_NOT_OPEN = "You can view the questions and any submitted responses for this feedback session but cannot submit new responses as the session is not currently open for submission.";
+        public static final String FEEDBACK_SUBMISSIONS_NOT_OPEN = "<strong>The feedback session is currently closed.</strong> You can view the questions and any submitted responses for this feedback session but cannot submit new responses.";
```
'not open for submission' is better? Sometimes sessions are opened by accident and then put back into awaiting state (which is technically not the same as 'closed')

damithc commented:
Minor change requested.

YongJieYongJie commented:
@wkurniawan07 Ready for review!

As requested, I changed the warning modal message body and message title:
 - `The feedback session is currently closed.` → `The feedback session is currently not open for submissions.`
 - `Feedback Session Closed` → `Feedback Session Not Open`
![image](https://cloud.githubusercontent.com/assets/14101781/15307456/afdfd334-1c05-11e6-9f26-cb73cd697d98.png)

wkurniawan07 commented:
Looks good :+1:
